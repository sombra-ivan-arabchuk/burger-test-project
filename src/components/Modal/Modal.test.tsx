import React from 'react';
import { render, RenderResult, waitForElement } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import CustomModal from './Modal';

export interface SetupProps {
  container: HTMLElement;
  utils: RenderResult;
}

const setup = (isOpened: boolean): SetupProps => {
  const utils = render(
    <CustomModal isOpen={isOpened} label={'test'}>
      <div>test text</div>
    </CustomModal>,
  );
  return {
    container: utils.container,
    utils,
  };
};

describe('Catalog', () => {
   it('check if modal is shown', async () => {
    const { utils } = setup(true);
    const { getByText } = utils;
    await waitForElement(() => getByText('test text'));
  });
});
