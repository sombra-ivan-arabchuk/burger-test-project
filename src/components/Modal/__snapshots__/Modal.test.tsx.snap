// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog renders 1`] = `null`;
